tests/contrib/boto/test.py

```diff
@@ -0,0 +1,120 @@
+# stdlib
+import unittest
+
+# 3p
+from nose.tools import eq_
+import boto.ec2
+import boto.s3
+import boto.awslambda
+import boto.sqs
+from moto import mock_s3, mock_ec2, mock_lambda
+
+# project
+from ddtrace import Pin
+from ddtrace.contrib.boto.patch import patch
+from ddtrace.ext import http
+
+# testing
+from ...test_tracer import get_dummy_tracer
+
+
+class BotoTest(unittest.TestCase):
+    """Botocore integration testsuite"""
+
+    TEST_SERVICE = "test-boto-tracing"
+
+    def setUp(self):
+        patch()
+
+    @mock_ec2
+    def test_ec2_client(self):
+        ec2 = boto.ec2.connect_to_region("us-west-2")
+        tracer = get_dummy_tracer()
+        writer = tracer.writer
+        Pin(service=self.TEST_SERVICE, tracer=tracer).onto(ec2)
+
+        ec2.get_all_instances()
+        spans = writer.pop()
+        assert spans
+        span = spans[0]
+        eq_(span.get_tag('aws.operation'), "DescribeInstances")
+        eq_(span.get_tag(http.STATUS_CODE), "200")
+        eq_(span.get_tag(http.METHOD), "POST")
+        eq_(span.get_tag('aws.endpoint'), "ec2")
+        eq_(span.get_tag('aws.region'), "us-west-2")
+
+        # Create an instance
+        ec2.run_instances(21)
+        spans = writer.pop()
+        assert spans
+        span = spans[0]
+        eq_(span.get_tag('aws.operation'), "RunInstances")
+        eq_(span.get_tag(http.STATUS_CODE), "200")
+        eq_(span.get_tag(http.METHOD), "POST")
+        eq_(span.get_tag('aws.endpoint'), "ec2")
+        eq_(span.get_tag('aws.region'), "us-west-2")
+
+        # Testing resource and service
+        eq_(span.service, "test-boto-tracing.ec2")
+        eq_(span.resource, "RunInstances.ec2.us-west-2")
+
+    @mock_s3
+    def test_s3_client(self):
+        s3 = boto.s3.connect_to_region("us-east-1")
+        tracer = get_dummy_tracer()
+        writer = tracer.writer
+        Pin(service=self.TEST_SERVICE, tracer=tracer).onto(s3)
+
+        s3.get_all_buckets()
+        spans = writer.pop()
+        assert spans
+        span = spans[0]
```

Comment:
one span is generated each time the function patched is called. For each request of the S3 client https://github.com/boto/boto/blob/develop/boto/s3/connection.py#L440, patched function gets called only once. On tests on our staging env there's a unique trace per s3 call. Yet in the mocking env I get 3 spans, I'll investigate on that